lib/puppet/pops/types/p_object_type.rb

```diff
@@ -333,6 +372,86 @@ def initialize(name_or_i12n_hash, i12n_hash_expression = nil)
     end
   end
 
+  # @api private
+  def new_function(loader)
+    @new_function ||= create_new_function(loader)
+  end
+
+  # @api private
+  def create_new_function(loader)
```

Comment:
I think this is problematic. The only documentation that can be added at this point would be documentation describing how a `new` function interacts with the meta-type `Object`. At present, the documentation is all about type conversions. Adding meta-type documentation at the same level will be confusing. It needs to be split up somehow.